src/Ast/AST.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types=1);
+
+namespace Google\Generator\Ast;
+
+use Google\Generator\Collections\Vector;
+
+/** Base of the PHP code AST. */
+abstract class AST
```

Comment:
Shouldn't casing be consistent between namespace and class name? I am not against it, as i don't mind the difference between class and namespace casing, but I believe it's more common to be consistent.

And the use of `AST` instead of `Ast` is correct according to [this naming RFC](https://wiki.php.net/rfc/class-naming)

Reply:
I'm not sure it matters too much here - I prefer how reads as all-caps in this specific case, but the namespace looks better to me as Pascal-cased!
I'd probably rather leave this as-is for now; we can always update it later if it's becoming inconsistent with other code in this project.

Reply:
It looks better to me also, I think it works, but wanted to call it out since it's not standard AFAIK